src/java.base/unix/classes/sun/nio/fs/UnixFileSystemProvider.java

```diff
@@ -358,21 +383,33 @@ public boolean isSameFile(Path obj1, Path obj2) throws IOException {
         if (!(obj2 instanceof UnixPath file2))
             return false;
 
-        UnixFileAttributes attrs1;
-        UnixFileAttributes attrs2;
+        // try to retrieve attributes following links
+        UnixFileAttributes attrs1 = null;
+        UnixFileAttributes attrs2 = null;
         try {
-             attrs1 = UnixFileAttributes.get(file1, true);
+            if ((attrs1 = UnixFileAttributes.getIfExists(file1)) != null)
+                attrs2 = UnixFileAttributes.getIfExists(file2);
         } catch (UnixException x) {
-            x.rethrowAsIOException(file1);
+            x.rethrowAsIOException(file1, file2);
```

Comment:
I think it would be better to handle them separately as that will give a better exception if either fails. With the 2-arg rethrowAsIOException is mostly for move/copy ops.

Reply:
Please see 78f272b.